frontend/model/captureLogs.js

```diff
@@ -1,218 +1,87 @@
   setItem('config', config)
 
-  setAppLogsFilter(sbp('state/vuex/state').settings?.appLogsFilter ?? [])
+  logger.setAppLogsFilter(sbp('state/vuex/state').settings?.appLogsFilter ?? [])
 
   // Subscribe to `appLogsFilter` changes.
-  sbp('okTurtles.events/on', SET_APP_LOGS_FILTER, setAppLogsFilter)
+  sbp('okTurtles.events/on', SET_APP_LOGS_FILTER, logger.setAppLogsFilter)
 
   // Overwrite the original console.
-  window.console = consoleCopy
+  self.console = logger.console
 
   // Set a new visit or session - useful to understand logs through time.
   // NEW_SESSION -> The user opened a new browser or tab.
   // NEW_VISIT -> The user comes from an ongoing session (refresh or login).
   const isNewSession = !sessionStorage.getItem('NEW_SESSION')
   if (isNewSession) { sessionStorage.setItem('NEW_SESSION', '1') }
-  console.log(isNewSession ? 'NEW_SESSION' : 'NEW_VISIT', 'Starting to capture logs of type:', appLogsFilter)
+  originalConsole.log(isNewSession ? 'NEW_SESSION' : 'NEW_VISIT', 'Starting to capture logs of type:', logger.appLogsFilter)
 }
 
-function captureLogsPause ({ wipeOut }: { wipeOut: boolean }): void {
-  if (wipeOut) { clearLogs() }
+async function captureLogsPause ({ wipeOut }: { wipeOut: boolean }): Promise<void> {
+  if (wipeOut) { await clearLogs() }
   sbp('okTurtles.events/off', SET_APP_LOGS_FILTER)
   console.log('captureLogs paused')
   // Restore original console behavior.
-  window.console = originalConsole
+  self.console = originalConsole
 }
 
-function clearLogs () {
-  removeItem('entries')
-  logger?.entries?.clear()
+async function clearLogs () {
+  await logger?.clear()
 }
 
-// Util to download all stored logs so far.
-function downloadOrShareLogs (actionType: 'share' | 'download', elLink?: HTMLAnchorElement): any {
-  const filename = 'gi_logs.json.txt'
-  const mimeType = 'text/plain'
-
-  const blob = new Blob([JSON.stringify({
-    // Add instructions in case the user opens the file.
-    _instructions: 'GROUP INCOME - Application Logs - Attach this file when reporting an issue: https://github.com/okTurtles/group-income/issues',
-    ua: navigator.userAgent,
-    logs: getLogger().entries.toArray()
-  }, undefined, 2)], { type: mimeType })
-
-  if (actionType === 'download') {
-    if (!elLink) { return }
-
-    const url = URL.createObjectURL(blob)
-    elLink.href = url
-    elLink.download = filename
-    elLink.click()
-    setTimeout(() => {
-      elLink.href = '#'
-      URL.revokeObjectURL(url)
-    }, 0)
-  } else {
-    return window.navigator.share({
-      files: [new File([blob], filename, { type: blob.type })],
-      title: L('Application Logs')
-    })
-  }
-}
+// The reason to use the 'visibilitychange' event over the 'beforeunload' event
+// is that the latter is unreliable on mobile. For example, if a tab is set to
+// the background and then closed, the 'beforeunload' event may never be fired.
+// Furthermore, 'beforeunload' has implications for how the bfcache works.
```

Comment:
`bfcache`? What's that?

Reply:
Back/forward cache: https://web.dev/articles/bfcache

Reply:
Please either include this link or add a parenthetical explaining what this is referring to since most people don't know